From df568f3bbbec7e316ab8df3d8ba08d3bfcc76461 Mon Sep 17 00:00:00 2001
From: Jianxun Zhang <jianxun.zhang@intel.com>
Date: Wed, 17 Apr 2013 15:19:19 -0700
Subject: Fix parsing string parameters in BOARD_MKBOOTIMG_ARGS

The existing logic in common.py breaks string arguments incorrectly:
e.g. --para1 val1 --para2 "val2 is a string" will be output as:

'--para', 'val1, '--para2', 'val2' 'is' 'a' 'string'

This will cause mkbootimg command fails due to the invalid arguments
generated from the wrong parsing.

The patch fixes this issue to get:

'--para', 'val1, '--para2', 'val2 is a string'

For: AXIA-1715
Change-Id: Ia34ec357550f11ae9d6adc719d86a0c6a9099fbc
Signed-off-by: Jianxun Zhang <jianxun.zhang@intel.com>
---
 core/Makefile                |    2 +-
 tools/releasetools/common.py |    3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index c33f607..00eb01c 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1258,7 +1258,7 @@ endif
 ifdef PRODUCT_EXTRA_RECOVERY_KEYS
 	$(hide) echo "extra_recovery_keys=$(PRODUCT_EXTRA_RECOVERY_KEYS)" >> $(zip_root)/META/misc_info.txt
 endif
-	$(hide) echo "mkbootimg_args=$(BOARD_MKBOOTIMG_ARGS)" >> $(zip_root)/META/misc_info.txt
+	$(hide) echo 'mkbootimg_args=$(BOARD_MKBOOTIMG_ARGS)' >> $(zip_root)/META/misc_info.txt
 	$(call generate-userimage-prop-dictionary, $(zip_root)/META/misc_info.txt)
 	@# Zip everything up, preserving symlinks
 	$(hide) (cd $(zip_root) && zip -qry $(ANDROID_BUILD_TOP)/$@ .)
diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index fa8a81e..2836fec 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -27,6 +27,7 @@ import tempfile
 import threading
 import time
 import zipfile
+import shlex
 
 try:
   from hashlib import sha1 as sha1
@@ -309,7 +310,7 @@ def BuildBootableImage(sourcedir, fs_config_file, info_dict=None):
 
   args = info_dict.get("mkbootimg_args", None)
   if args and args.strip():
-    cmd.extend(args.split())
+    cmd.extend(shlex.split(args))
 
   cmd.extend(["--ramdisk", ramdisk_img.name,
               "--output", img.name])
-- 
1.7.4.1

